bciers/apps/registration1/tests/utils/updateDashboardDataHref.test.ts

```diff
@@ -0,0 +1,72 @@
+import { describe, expect, it } from "vitest";
+import updateDashboardDataHref from "@bciers/utils/updateDashboardDataHref";
+import { ContentItem } from "@bciers/types/tiles";
+
+const items: ContentItem[] = [
+  {
+    icon: "icon1",
+    href: "/dashboard/item1",
+    title: "Dashboard Item 1",
+    content: "This is item 1",
+    links: [{ href: "/link1", title: "Link 1", conditions: [] }],
+    conditions: [],
+  },
+  {
+    icon: "icon2",
+    href: "/dashboard/item2",
+    title: "Dashboard Item 2",
+    content: "This is item 2",
+    links: [{ href: "/link2", title: "Link 2", conditions: [] }],
+    conditions: [],
+  },
+];
+
+describe("updateDashboardDataHref function", () => {
+  it("should update href by replacing static value and appending title", () => {
+    const replaceValue = "/dashboard";
+    // Mock functions for getNewValue and getTitle
+    const getNewValue = (item: ContentItem) => `/newpath/${item.title}`;
+    const getTitle = (item: ContentItem) => `?title=${item.title}`;
+
+    // Call the function with the mocks
+    const updatedItems = updateDashboardDataHref(
+      items,
+      replaceValue,
+      getNewValue,
+      getTitle,
+    );
+
+    // Expectations for the updated items
+    expect(updatedItems).toEqual([
+      {
+        icon: "icon1",
+        href: "/newpath/Dashboard Item 1/item1?title=Dashboard Item 1",
+        title: "Dashboard Item 1",
+        content: "This is item 1",
+        links: [{ href: "/link1", title: "Link 1", conditions: [] }],
+        conditions: [],
```

Comment:
What are `conditions`? I don't see those in the function

Reply:
We are not checking conditions here, so I can remove them. I only tried to create close mock data for our tiles.